database/engine/rrdengine.c

```diff
@@ -1339,13 +1339,12 @@ void rrdeng_worker(void* arg)
  */
 void rrdengine_main(void)
 {
-    int ret;
-    struct rrdengine_instance *ctx;
+    STORAGE_ENGINE_INSTANCE* ctx;
 
     sanity_check();
-    ret = rrdeng_init(NULL, &ctx, "/tmp", RRDENG_MIN_PAGE_CACHE_SIZE_MB, RRDENG_MIN_DISK_SPACE_MB);
-    if (ret) {
-        exit(ret);
+    ctx = rrdeng_init(storage_engine_get(RRD_MEMORY_MODE_DBENGINE), NULL);
```

Comment:
Hello @aberaud ,

The `rrdeng_init` function is calling [callocz](https://github.com/netdata/netdata/blob/f62bc0efddb2a5ddaa11655bb59771478b11e44b/libnetdata/libnetdata.c#L175-L179) that will close netdata calling `fatal` when memory cannot be allocated, so next test calling `exit` is not necessary.

Reply:
Done